src/Controls/tests/Core.UnitTests/GridTests.cs

```diff
@@ -2394,6 +2394,17 @@ public void MinimumHeightRequestInAutoCells()
 			Assert.Equal(boxRow0Column1.MinimumHeightRequest, boxRow0Column1.Height);
 		}
 
+		[Fact]
+		public void ShouldPropagateVisibilityToChildren()
```

Comment:
Could we move this to VisualElementTests? Maybe also add the full combo of tests, with maybe 2 and 3 levels of nesting.

```cs
var root = new Grid();
var nested = new Grid();
var child = new Button();
```

and then have a matrix of how this should propagate.

```cs
[InlineData(false, true, true, false, false, false)]
public void IsVisiblePropagates(bool root, bool nested, bool child, bool expectedRoot, bool expectedNested, bool expectedChild)
```

This way we can confirm all the combinations are as expected and they are never going to break by some external logic.

Reply:
I would also add a test to make sure we can un-set:

```cs
var button = new Button();
var grid = new Grid { button };

grid.IsVisible = false;
Aseert.False(button.IsVisible);

grid.IsVisible = true;
Aseert.True(button.IsVisible);
```


I would also add a test to make sure we can un-set:

```cs
var button = new Button { IsVisible = false };
var grid = new Grid { button };

grid.IsVisible = false;
Aseert.False(button.IsVisible);

// button stays hidden if it was explicitly set
grid.IsVisible = true;
Aseert.False(button.IsVisible);
```


I would also add a test to make sure we can un-set:

```cs
var button = new Button { IsVisible = false };
var grid = new Grid { button };

// everything is hidden
grid.IsVisible = false;
Aseert.False(button.IsVisible);

// make buttin visible, but it should not appear
button.IsVisible = true;
Aseert.False(button.IsVisible);

// even when the grid is visible, the button is not
grid.IsVisible = true;
Aseert.False(button.IsVisible);
```

Reply:
Thanks! I will try to do it tomorrow

Reply:
Hi @mattleibow I've written the tests and they are passing except the last one. However, it failed when I replaced `IsVisible` with `IsEnabled`. In my opinion it is a correct behavior. In that way, we can prepare the element to be visible when a parent is visible. What do you think?

```
		public void ButtonShouldBeVisibleWhenExplicitlySetWhenParentIsVisible()
		{
			var button = new Button { IsVisible = false };
			var grid = new Grid { button };

			// everything is hidden
			grid.IsVisible = false;
			Assert.False(button.IsVisible);

			// make buttin visible, but it should not appear
			button.IsVisible = true;
			Assert.False(button.IsVisible);

			// button appears when parent appears
			grid.IsVisible = true;
			Assert.True(button.IsVisible);
		}
```

This test will pass with the current implementation

Reply:
Ah, maybe I had a type. Thanks for checking and doing the right thing :) Tests are cool so now it is working as expected.